### ISSUE TITLE
Uncaught (in promise) TypeError: Cannot read property 'localeCompare' of undefined when trying to add thing to model

### ISSUE THREAD

**Issue by austvik** [openhab/openhab-webui#1137]
## The problem

I have migrated my OpenHab from 2.5 to 3.0 and have just started to set up my semantic model. Adding (new) locations work fine, but when I try to get some equipment in, I get JavaScript errors (Edge and Chrome on Windows), and I am not able to select a Thing:

![image](https://user-images.githubusercontent.com/601917/127779048-71670923-254d-4861-874d-f4bd98afe257.png)

Uncaught (in promise) TypeError: Cannot read property 'localeCompare' of undefined
 
![image](https://user-images.githubusercontent.com/601917/127778960-b0c408c1-8d09-43bc-ab35-e503894c3101.png)

I guess it is data driven (maybe line 21), but I am not sure if it is the sort that is not robust enough or things that return illegal data

![image](https://user-images.githubusercontent.com/601917/127778976-4f35bc37-e43e-401c-a230-885396c01a6d.png)

The Settings/Things works though, I guess on the same data:

![image](https://user-images.githubusercontent.com/601917/127779008-9fd31710-3217-423e-a6d4-c8a8c017bc36.png)

The issue is both on "Create Equipment From Thing" and "Create Points from Thing".

## Expected behavior

I expect to be able to select an existing thing in the list and add it as equipment or point into the semantic model

## Steps to reproduce

1.  Get some strange data into things.
2. Try to sort based on it

## Your environment

```yaml
runtimeInfo:
  version: 3.1.0
  buildString: Release Build
locale: en-NO
systemInfo:
  configFolder: /etc/openhab
  userdataFolder: /var/lib/openhab
  logFolder: /var/log/openhab
  javaVersion: 11.0.12
  javaVendor: Azul Systems, Inc.
  javaVendorVersion: Zulu11.50+19-CA
  osName: Linux
  osVersion: 5.10.52-v7l+
  osArchitecture: arm
  availableProcessors: 4
  freeMemory: 81896744
  totalMemory: 194772992
bindings:
  - astro
  - deconz
  - denonmarantz
  - harmonyhub
  - hpprinter
  - somfytahoma
  - sonos
  - spotify
  - tibber
  - zigbee
  - zwave
clientInfo:
  device:
    ios: false
    android: false
    androidChrome: false
    desktop: true
    iphone: false
    ipod: false
    ipad: false
    edge: false
    ie: false
    firefox: false
    macos: false
    windows: true
    cordova: false
    phonegap: false
    electron: false
    nwjs: false
    webView: false
    webview: false
    standalone: false
    os: windows
    pixelRatio: 1.5
    prefersColorScheme: light
  isSecureContext: false
  locationbarVisible: true
  menubarVisible: true
  navigator:
    cookieEnabled: true
    deviceMemory: N/A
    hardwareConcurrency: 8
    language: en-GB
    languages:
      - en-GB
      - en
      - en-US
      - no
      - de
    onLine: true
    platform: Win32
  screen:
    width: 2560
    height: 1440
    colorDepth: 24
  support:
    touch: false
    pointerEvents: true
    observer: true
    passiveListener: true
    gestures: false
    intersectionObserver: true
  themeOptions:
    dark: light
    filled: true
    pageTransitionAnimation: default
    bars: filled
    homeNavbar: default
    homeBackground: default
    expandableCardAnimation: default
  userAgent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML,
    like Gecko) Chrome/92.0.4515.107 Safari/537.36 Edg/92.0.902.62
timestamp: 2021-08-01T16:53:44.468Z

```

## Browser console

![image](https://user-images.githubusercontent.com/601917/127779121-73278d58-6ab7-407a-ac29-d7661cb64c49.png)

## Browser network traffic

![image](https://user-images.githubusercontent.com/601917/127779147-5c945117-497d-47d6-bc4e-75befa4a01f4.png)

**Comment by ghys** [comment]
This looks similar to #762 (but in the things picker instead of the things list).
Sometimes thing labels would be omitted from the API response for some reason.
A workaround could be added (like #769) but it would be nice to find the root cause.
Check that you don't have things without a label (often defined in .things files) and try to stop, clean cache and restart openHAB.

**Comment by austvik** [comment]
Thanks for the tip!

This jq on the response from things helped me locate it: ".[].label"

Here is the element:

>     {
>        "statusInfo": {
>            "status": "UNINITIALIZED",
>            "statusDetail": "HANDLER_REGISTERING_ERROR",
>            "description": "class org.openhab.core.thing.internal.ThingImpl cannot be cast to class org.openhab.core.thing.Bridge (org.openhab.core.thing.internal.ThingImpl and org.openhab.core.thing.Bridge are in unnamed module of loader org.eclipse.osgi.internal.loader.EquinoxClassLoader @1beb23f)"
>        },
>        "editable": true,
>        "configuration": {},
>        "properties": {},
>        "UID": "zwave:serial_zstick:0d6dcafc",
>        "thingTypeUID": "zwave:serial_zstick",
>        "channels": []
>    },

**Comment by austvik** [comment]
How it looked in the things page (fell back to UID if label was missing):
![image](https://user-images.githubusercontent.com/601917/128071965-4bbcc4b2-c3b1-41d9-8626-b973913c420c.png)

**Comment by austvik** [comment]
I deleted it from the org.openhab.core.thing.Thing jsondb, and now it works:
![image](https://user-images.githubusercontent.com/601917/128073161-bb0ec9cb-105a-45c2-8034-38a7d49c7b7e.png)

**Comment by ghys** [comment]
Closed as fixed for the author and a duplicate of #762. There seems to be an underlying issue with missing labels in the API in some cases.
